### PR TITLE
Adding Python 3.4 support and notes on Python 3.6

### PR DESCRIPTION
**Pull request by jason-plus3**
Simply install the yum packages, pip, and then pip update itself and virtualenv for both Python 2.7 and 3.4.

Installation for Python 3.6 is documented, but relies on the semi-official IUS rpm repo.  Steps are commented out to avoid that dependency until it is specifically requested.

### CODE REVIEW

File: SupportFiles/jenkins-agent_userscript.sh [hunk 1]
```diff
@@ -51,6 +51,23 @@ printf 'Pulling agent.jar from %s' "${JENKMSTR}"
...
   err_exit "Failed to pull agent.jar"
 
+
+# Install Python 2.x, pip, and virtualenv
+yum install -y python python2-pip
+pip2 install -U virtualenv pip
+
+# Install Python3.4, pip, and virtualenv
+yum install -y python34 
+curl https://bootstrap.pypa.io/get-pip.py | python3.4  # no rpm in epel
```
Comment (lorengordon) [inline SupportFiles/jenkins-agent_userscript.sh]: IIRC from prior foolery, this overwrites the `pip` script installed by the RPM. Run `pip --version` to doublecheck... If that's not intended, flipping the install order ought to fix, i.e. handle py34/36 first, and py27 last

Comment (jason-plus3) [reply SupportFiles/jenkins-agent_userscript.sh]: Naively, my assumption would be that `pip3`, `pip2`, etc give you a specific version and that `pip` results in whatever the system decides to give you.  

To me, the proper solution would to delete the undefined behavior of `pip` and force use of a versioned pip.  But I don't get Python's ecosystem and welcome alternative resolutions.

Comment (lorengordon) [reply SupportFiles/jenkins-agent_userscript.sh]: I don't have a horse in this, and don't particularly care either way. I'm just identifying a potentially unintended side effect. If it's intended, or if it just doesn't matter for this system for whatever reason, feel free to ignore me.

**Comment by ferricoxide** [comment]
Out of date .travis.yml. Travis has changed their test-environment since the last time successful tests were run. Get rid of the:

~~~~
addons:
  apt:
    sources:
      - debian-sid    # Grab ShellCheck from the Debian repo
    packages:
      - shellcheck
~~~~

Block and it will probably work.

**Pull request merged**